### PR TITLE
add proposal to specify lvcreate options on SC

### PR DESCRIPTION
**Pull request by toshipp**
To solve the problem where users can not different lv create options on same VG (#625), add a new parameter to specify the options on SC.

### CODE REVIEW

**Review by peng225 — COMMENTED** [review]
LV create options was once discussed in the following proposal.
https://github.com/topolvm/topolvm/blob/main/docs/proposals/lvcreate-options.md

It may be helpful for readers to refer to the old proposal and state how our previous dicision was wrong.

**Review by cupnes — DISMISSED** [review]

File: docs/proposals/lvcreate-option-on-storageclass.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+Change lvmd.conf format as bellow:
+
+```yaml
+socket-name: /run/topolvm/lvmd.sock
+device-classes:
+  - name: dc
+    volume-group: vg
+    spare-gb: 10
+lvcreate-option-class:
```
Comment (daichimukai) [inline docs/proposals/lvcreate-option-on-storageclass.md:39]:
```suggestion
lvcreate-option-classes:
```

**Review by daichimukai — DISMISSED** [review]

File: docs/proposals/lvcreate-option-on-storageclass.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+The parameter value is passed to the topolvm-controller via CreateVolume RPC.
+We have to modify LogicalVolume CR to pass the value from the controller to the topolvm-node
+and also modify lvmd's CreateLVRequest RPC.
+
+### Migration
+
+The new parameter supersedes `stripe`, `stripe-size`, and `lvcreate-options` on device-class,
+we deprecate these parameters and will remove them in the feature.
+But this breaks current behavior, we use these parameters if and only if the new parameter is not specified.
```
Comment (satoru-takeuchi) [inline docs/proposals/lvcreate-option-on-storageclass.md:75]:
```suggestion
Before the removal of these deprecated parameters, we use these parameters if and only if the new parameter is not specified.
```

File: docs/proposals/lvcreate-option-on-storageclass.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# LV create options on StorageClass
+
+## Summary
+
+Allow specifying LV create options on StorageClass parameter.
+
+## Motivation
+
+We want to specify different LV create options for a VG, however, it is not
```
[the remark this answers is not in the document]
Comment (toshipp) [reply docs/proposals/lvcreate-option-on-storageclass.md:9]: In this situation, we will make two or more LVs from one VG, and each LV will be made by using a different create option.
So is it correct to use `each`?

**Review by peng225 — APPROVED** [review]

File: docs/proposals/lvcreate-option-on-storageclass.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (satoru-takeuchi) [reply docs/proposals/lvcreate-option-on-storageclass.md:9]: Ah, indeed. Please ignore this suggestion.

**Review by satoru-takeuchi — APPROVED** [review]

**Comment by toshipp** [comment]
I missed minor problem. I'll fix it later.

**Pull request merged**